quakemigrate/plot/phase_picks.py

```diff
@@ -70,87 +74,163 @@ def pick_summary(event, station, signal, picks, onsets, ttimes, window):
+                bbox=dict(boxstyle="round", fc="w", alpha=0.8), va="top",
+                ha="left", fontsize=18, zorder=2)
+        # Set ylim
+        y_min = min(y[min_win_idx:max_win_idx])
+        y_max = max(y[min_win_idx:max_win_idx])
+        ax.set_ylim(ymin=(y_min - 0.1 * abs(y_min)),
+                    ymax=(y_max + 0.1 * abs(y_max)))
+
+    # --- Plot onset functions ---
+    # Handle case where only S phase is used
+    n = 3
+    if len(phases) == 1 and phases[0] == "S":
+        n += 1
+
+    # Loop through all relevant onset function axes
+    for i, (ax, ph) in enumerate(zip(axes[n:5], phases)):
+        # Plot onset functions
         y = onsets[i]
-        win = window[ph]
-        # Get normalising factor within window
-        norm = np.max(abs(y[win[0]:win[2]+1]))
-        ax.plot(times, y / norm, c="k", lw=0.5, zorder=1)
-
-    for ax in axes:
-        ax.set_xlim([min_t.datetime, max_t.datetime])
-
-    # --- Plot labels and fix limits ---
-    shift = (max_t - min_t) * 0.01  # Fractional shift for text label
-    for comp, ax in zip(["Z", "N", "E"], axes[:3]):
-        ax.text((min_t+shift).datetime, 0.9, f"{station}.BH{comp}",
-                ha="left", va="center", zorder=2, fontsize=18)
-        ax.set_ylim([-1.1, 1.1])
-        ax.set_yticks(np.arange(-1, 1.5, 0.5))
-    for ph, ax in zip(["P", "S"], axes[3:]):
-        ax.text((min_t+shift).datetime, 1., f"{ph} onset", ha="left",
-                va="center", zorder=2, fontsize=18)
-        ax.set_ylim([-0.1, 1.1])
-        ax.set_yticks(np.arange(0., 1.2, 0.2))
+        ax.plot(times[min_idx:max_idx+1], y[min_idx:max_idx+1], c="k", lw=0.5,
+                zorder=1)
+
+        # Plot labels
+        ax.text(0.015, 0.95, f"{ph} onset", transform=ax.transAxes,
+                bbox=dict(boxstyle="round", fc="w", alpha=0.8), va="top",
+                ha="left", fontsize=18, zorder=2)
+
+        # Plot threshold
+        gau = event.picks["gaussfits"][station][ph]
+        thresh = gau["PickThreshold"]
+        ax.axhline(thresh, label="Pick threshold")
+        text.text(0.05+i*0.5, 0.2, f"Threshold: {thresh:5.3f}", ha="left",
+                  va="center", fontsize=18)
+
+        # Plot gaussian fit to onset function
+        if not gau["PickValue"] == -1:
+            yy = util.gaussian_1d(gau["xdata"], gau["popt"][0],
+                                  gau["popt"][1], gau["popt"][2])
+            dt = [x.datetime for x in gau["xdata_dt"]]
+            ax.plot(dt, yy)
+
+        # Set ylim
+        win = windows[ph]
+        onset_max = max(onsets[i][win[0]:win[2]+1])
+        y_max = max(onset_max, thresh)
+        ax.set_ylim(ymin=0, ymax=y_max*1.1)
 
     # --- Plot predicted arrival times ---
-    for i, ax in enumerate(axes):
-        model_pick = otime + ttimes[0] if i % 3 == 0 else otime + ttimes[-1]
+    # Handle case where only a single phase is used
+    ax_ind = [ind for ind in range(5)]
```

Comment:
Is the list comprehension necessary? `range` returns an iterator, not a generator, so it doesn't get 'used' up, so I think just `ax_ind = range(5)` should be safe?

Reply:
```suggestion
    ax_ind = range(5)
```